### ISSUE TITLE
Wiki pages we should have :)

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/sqlitebrowser#226]
Placeholder issue for wiki pages we should add:

* How to use regular expressions in the Execute SQL tab, as [added here](https://github.com/sqlitebrowser/sqlitebrowser/commit/6d1a874f8077b859bee2dd9d4031f242cfa1e807) by @MKleusberg.

Anyone wanting to help out is definitely welcome (and encouraged!) to being or improve any of these pages in the wiki! :grinning:

**Comment by samir-aguiar** [comment]
Added one more item :grinning:

**Comment by justinclift** [comment]
Good thinking.  And yeah, we should add stuff here as we think of it. :smile:

**Comment by MKleusberg** [comment]
Just added one more, too :)

**Comment by Z4us** [comment]
Just corrected a page :)

**Comment by glawrence** [comment]
@justinclift I have written a wiki page on Regular Expressions at https://github.com/sqlitebrowser/sqlitebrowser/wiki/Regular-Expressions and update the wiki home page. Please feel free to improve as you see necessary.

**Comment by justinclift** [comment]
@glawrence You really are pretty amazing.  That's truly awesome. :grinning:

I've just tweaked the wording a bit, and added a screenshot for clarity too.  (turns out adding screenshots isn't straightforward, but it is possible)

**Comment by glawrence** [comment]
@justinclift you're welcome. I am glad you tweaked the wording, my writing usually needs a tweak or two, so thank you. I will try and remember to add screenshots next time, they do help.

**Comment by justinclift** [comment]
Added an item for making a wiki page to explain what our project files are for.  As I don't seem to have gotten around to it in https://github.com/sqlitebrowser/sqlitebrowser/issues/35. :wink:

**Comment by justinclift** [comment]
Added an item for "How to create an encrypted database (and/or encrypt an existing one)".  @glawrence Any interest in that? :smile:

**Comment by glawrence** [comment]
Seems we both had the same thought! Let me know if what I have done does not cover it enough, very happy to have it updated/corrected/improved.

**Comment by justinclift** [comment]
@glawrence Definitely needs some screenshots added.  Do you have something that runs the SQLCipher versions and can do them with?

**Comment by glawrence** [comment]
@justinclift my default OS is Windows, so no. I have a Linux VM but have not built with SQLCipher yet, currently working on something else. I also don't have your image editing skills with those soft borders you add!

**Comment by justinclift** [comment]
Ahhh, the soft borders is an automatic thing done by the inbuilt OSX screenshot key combo.  :wink:

I'll get some screenshots done for the SQLCipher stuff done at some point, as I think it'll help. :grin:

**Comment by glawrence** [comment]
I agree, it would help. To be fair images have always been an Apple strong point.

**Comment by glawrence** [comment]
I am working on documenting the Windows build process, which will start as a Wiki page. Some issues to refer to are as follows:

[Open Encrypted Database #12] (https://github.com/sqlitebrowser/sqlitebrowser/issues/12)
[Complete quoting for parameters of some CMake commands #58] (https://github.com/sqlitebrowser/sqlitebrowser/issues/58)
[Build sqllitebrowser from source, sqlcipher support (Windows) #482](https://github.com/sqlitebrowser/sqlitebrowser/issues/482)

**Comment by revolter** [comment]
Any update on the build wiki? Also, will it apply to OSX too? I want to contribute but I'm not able to build and run the project. And I don't know how to use Qt creator. I would want to know how to open an IDE and view the source code, build it and run it. I created a Qt project from the `CMakeLists.txt` file but didn't see any source code.

**Comment by MKleusberg** [comment]
One more thing: We need to update the wiki page on using the filters. There are two things that changed in the meantime:
1) We've added a range operator (```1~5```). See #939.
2) We've changed the default behaviour when no operator is specified. In the Wiki it says that for text a case-independent exact search is performed by default. But that isn't true anymore. What we do now is add ```%``` in the front and at the end of the search string. So filtering for ```b``` actually queries ```%b%```. To override this you have to use the ```=``` operator (```=b```) or manually add ```%``` characters: ```b%``` doesn't expand to ```%b%%`.

**Comment by MKleusberg** [comment]
Just added a new wiki page listing the more hidden keyboard shortcuts that I know of: https://github.com/sqlitebrowser/sqlitebrowser/wiki/Keyboard-shortcuts Feel free to extend it :smiley:

**Comment by mgrojo** [comment]
This is a suggestion from #821:
- "How to use the Plot pane" (with screenshots)

**Comment by sky5walk** [comment]
Please add a tooltip for Prefetch block size.
The ? tool does not work within the preferences dialog.

I got burned by the following:
Execute SQL - Select all resulting rows&cols - Copy to clipboard.
Result = Only the prefetch amount of rows were selected and copied.
If the user sets their prefetch to a small number for performance, they need to know the ramifications.

**Comment by justinclift** [comment]
@sky5walk Would you be ok to create a new Issue for that request.  It doesn't seem like a wiki page thing, which which Issue is for. :smile:

**Comment by mgrojo** [comment]
Any help for updating the wiki pages for the 3.11 release is welcome. I've added two new pages and updated some others to be in line with the new release:
- [X] [Display Formats](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Display-Formats)
- [X] [Plot Dock](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Plot-Dock)

I think these are most needed pages for the next release:
- [ ] Projects - How and when to use project files
- [ ] Views - Working with views, ways to create them and unlocking view editing.
- [ ] Preferences - Available configuration options. This could be the basis of an in-line what's-this information that could be added to the application (that has been requested more than once).
- [ ] Exporting - Available formats and options for exporting data and schema
- [ ] Importing - Available formats and options for importing data and schema

**Comment by chrisjlocke** [comment]
I'll add a couple.  A good list there.

**Comment by mgrojo** [comment]
Added some to the list above:
- Remote panel and integration to dbhub.io
- SQL/Error Log
- Help for every important dialog, like:
  - Edit table
  - Edit index
  - Add record

**Comment by scottfurry** [comment]
Almost as if we need a dedicated Help manual... :thinking: :thinking:

**Comment by justinclift** [comment]
That's not a bad idea. :smile:

**Comment by scottfurry** [comment]
With all the issues reports and user problems - a Libreoffice-like (manual is almost needed).
Chapter 1 -> Installation.
Chapter 2 -> Opening a database
....

**Comment by chrisjlocke** [comment]
Or just lessons on the Wiki, similar to this:
https://www.vbtutor.net/lesson17.html

with complete 'walkthroughs'.

**Comment by houghtonap** [comment]
I would add a list of the built-in extensions in DB4S and whether they were compiled with ICU compatibility for international support.

**Comment by justinclift** [comment]
> I would add a list of the built-in extensions in DB4S and whether they were compiled with ICU compatibility for international support.

That's a good idea.  If it helps in the meantime, this the compile line used for the Win64 builds (Win32 is identical):

https://github.com/sqlitebrowser/sqlitebrowser/blob/acd23e1fc36289818c7b07401aae487844bfb5de/installer/windows/nightly_build_scripts/win64build.bat#L70
